python/ray/ml/preprocessor.py

```diff
@@ -87,13 +123,17 @@ def transform_batch(self, df: DataBatchType) -> DataBatchType:
         Returns:
             DataBatchType: The transformed data batch.
         """
-        if self._is_fittable and not self.check_is_fitted():
+        fit_status = self.fit_status()
+        if fit_status in (
+            Preprocessor.FitStatus.PARTIALLY_FITTED,
+            Preprocessor.FitStatus.NOT_FITTED,
+        ):
             raise PreprocessorNotFittedException(
                 "`fit` must be called before `transform_batch`."
             )
         return self._transform_batch(df)
 
-    def check_is_fitted(self) -> bool:
+    def _check_is_fitted(self) -> bool:
```

Comment:
From other recent suggestions, it's probably easier to use this as a property:

```suggestion
    @property
    def _is_fitted(self) -> bool:
```

Reply:
prefer to keep it this way so not to be confused with `fit_status()`, which is actually more acting like property in this case.